src/ui/svgWidget.js

```diff
@@ -0,0 +1,78 @@
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Create a new instance of class geo.gui.svgWidget
+ *
+ * Due to the nature of d3 creating DOM elements as it inserts them, calls to appendChild
+ * don't appear in this widget.
+ *
+ * The canvas of an svgWidget always refers to the actual svg element.
+ * The parentCanvas can refer to another widgets svg element, dom element, or the
+ * UI layers dom element.
+ * See {@link geo.gui.widget#parentCanvas}.
+ *
+ * @class
+ * @extends geo.gui.domWidget
+ * @returns {geo.gui.svgWidget}
+ *
+ */
+//////////////////////////////////////////////////////////////////////////////
+geo.gui.svgWidget = function (arg) {
+  'use strict';
+  if (!(this instanceof geo.gui.svgWidget)) {
+    return new geo.gui.svgWidget(arg);
+  }
+
+  geo.gui.domWidget.call(this, arg);
+
+  var m_this = this,
+      m_renderer = null;
+
+  this._init = function (arg) {
+    var d3Parent;
+    arg = m_this.parseArgs(arg);
+    m_this.args = arg;
```

Comment:
I would rather get rid of `widget.prototype.parseArgs`, move its code into `widget.prototype._init`, and then call that from here.  This should be calling the superclass `_init` anyway.  You can either place the options directly on `m_this` or add a get/set method for each like we do in [other classes](https://github.com/OpenGeoscience/geojs/blob/master/src/core/layer.js#L101).